crates/ty_server/src/server/api.rs

```diff
@@ -177,8 +223,20 @@ fn background_notification_thread<'a, N: traits::BackgroundDocumentNotificationH
             return Box::new(|_, _| {});
         };
         Box::new(move |notifier, _| {
-            let _span = tracing::trace_span!("notification", method = N::METHOD).entered();
-            if let Err(err) = N::run_with_snapshot(snapshot, notifier, params) {
+            let _span = tracing::debug_span!("notification", method = N::METHOD).entered();
+
+            let result = match ruff_db::panic::catch_unwind(|| {
+                N::run_with_snapshot(snapshot, notifier, params)
+            }) {
+                Ok(result) => result,
+                Err(panic) => {
+                    tracing::error!("An error occurred while running {id}: {panic}");
+                    show_err_msg!("ty encountered a problem. Check the logs for more details.");
+                    return;
+                }
```

Comment:
I'd recommend updating the message to highlight this is related to the "panic" instead of an error. It would be useful when looking at the log messages to understand the origin of the message.